src/client/Microsoft.Identity.Client/Platforms/Features/OpenTelemetry/OtelInstrumentation.cs

```diff
@@ -136,26 +144,28 @@ public OtelInstrumentation()
             }
 
             // Only log duration in microseconds when the cache level is L1.
-            if (s_durationInL1CacheInUs.Value.Enabled && authResultMetadata.TokenSource.Equals(TokenSource.Cache) 
+            if (s_durationInL1CacheInUs.Value.Enabled && tokenSource == TokenSource.Cache
                 && authResultMetadata.CacheLevel.Equals(CacheLevel.L1Cache))
             {
                 s_durationInL1CacheInUs.Value.Record(totalDurationInUs,
                 new(TelemetryConstants.MsalVersion, MsalIdHelper.GetMsalVersion()),
                 new(TelemetryConstants.Platform, platform),
                 new(TelemetryConstants.ApiId, apiId),
-                new(TelemetryConstants.TokenSource, authResultMetadata.TokenSource),
+                new(TelemetryConstants.TokenSource, tokenSource),
                 new(TelemetryConstants.CacheLevel, cacheLevel));
             }
         }
 
-        void IOtelInstrumentation.LogFailedMetrics(string platform, string errorCode)
+        void IOtelInstrumentation.LogFailedMetrics(string platform, string errorCode, ApiEvent.ApiIds apiId, bool isProactiveTokenRefresh)
```

Comment:
nit: why did you use explicit interface implemenation? We should default to the non-explicit https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/interfaces/explicit-interface-implementation

Reply:
Good point, this is not intentional. Might be some intellisense or copilot thing that went unnoticed.